docker/pulsar/Dockerfile

```diff
@@ -33,7 +33,13 @@ COPY scripts/pulsar-zookeeper-ruok.sh /pulsar/bin
 COPY scripts/watch-znode.py /pulsar/bin
 COPY scripts/install-pulsar-client.sh /pulsar/bin
 
-RUN mkdir /pulsar/data
+# In order to support running this docker image as a container on OpenShift
+# the final image needs to give the root group sufficient permission.
+# The file permissions are preserved when copying files from this builder image to the target image.
+RUN chmod -R g+w /pulsar/conf
+RUN mkdir /pulsar/data && chmod -R g+w /pulsar/data
+RUN mkdir /pulsar/logs && chmod -R g+w /pulsar/logs
+RUN chmod -R u-w /pulsar
```

Comment:
I think that the number of RUN lines in the Dockerfile should be minimized, since each RUN will create a new layer. All of these lines could be combined in one RUN.
`docker build --squash`/`-Ddockerfile.build.squash=true` resolves that issue for the final docker image, however  that would require enabling experimental mode in the docker daemon which isn't there by default ([example for GH CI](https://github.com/lhotari/pulsar/blob/178eebdd97c6b9fff6b460089bdc3d8d5e4337a3/.github/actions/tune-os/action.yml#L44-L47)).

Reply:
typically something like `chmod -R u-w /pulsar` is very expensive from diskspace consumption perspective since Docker's copy-on-write (COW) solution must copy the file also when the file permissions are modified. This would double the disk space if /pulsar contained the files. However, it seems that files are copied to /pulsar after this command.

Reply:
Now, when I looked at the full diff, I noticed that an intermediate container image is used, so the waste in the disk space doesn't really matter that much. I guess it will happen under the covers that 2x diskspace will be consumed as a result of `chmod -R u-w /pulsar` call. That won't carry on to the final Docker image since this an intermediate container image.

Reply:
Yes, it is less efficient, but since it is a builder image, I think it is fine. My goal on this line is to give only the necessary permission to the container.